DailyKit/CallManager/CallManager.swift

```diff
@@ -0,0 +1,134 @@
+import Foundation
+import Combine
+import Daily
+
+public final class CallManager: CallManageable, CallClientDelegate {
+    public var url: URL? { subjects.url.value }
+    public var callState: CallState { subjects.callState.value }
+    public var camera: CallCamera { subjects.camera.value }
+    public var microphone: CallMicrophone { subjects.microphone.value }
+    public var participants: CallParticipants { subjects.participants.value }
+
+    private let callClient: CallClient
+    private let subjects: CallClientSubjects
+    private var participantsBuilder: CallParticipants.Builder
+
+    public convenience init() {
+        self.init(callClient: CallClient())
+    }
+
+    required init(callClient: CallClient) {
+        self.callClient = callClient
+        self.participantsBuilder = CallParticipants.Builder(
+            local: callClient.participants.local.asCallParticipant
+        )
+        self.subjects = CallClientSubjects(
+            url: callClient.url,
+            callState: callClient.callState,
+            camera: CallCamera(callClient.inputs.camera),
```

Comment:
The individual subjects are [`CurrentValueSubject<T, E>`](https://developer.apple.com/documentation/combine/currentvaluesubject), i.e. streams that cache their most-recent received value.

The properties (getters) provide an ergonomic and synchronous access to individual "current" states (imperative programming), while the subjects allow for attaching event handlers to the updates (reactive programming). The properties aren't strictly necessary as the user could just as well access the subject's value themselves, but they make the code a bit cleaner and nicer to read.

A lot of the code in this starter kit is likely to get upstreamed into `daily-ios` at some point in some form or another. So in some semi-distant future the necessity for such user-provided wrappers (e.g. `CallManager`, `CallParticipants`, …) will hopefully go away entirely.

Reply:
> Is its purpose just to hold the publishers collection?

A common type for the publishers is the main purpose. I wanted to avoid duplicating this code in both implementations of `CallManageable`, especially with the casting / `fatalError` part. As @regexident mentioned, we're thinking about adding something like `func publisher<T>(for property: CallClientProperty<T>) -> AnyPublisher<T, Never>` to the `CallClient`, so this was also an effort to start thinking through a general abstraction we might want to use there too.